### PR TITLE
Spring-05

### PR DESCRIPTION
**Pull request by Max-Pochepets**
REST

- Created DTO for CinemaHall, Movie and MovieSession
- Created controller for respective model

### CODE REVIEW

File: src/main/java/cinema/shop/service/impl/CinemaHallServiceImpl.java [hunk 1]
```diff
@@ -19,6 +19,12 @@ public CinemaHall add(CinemaHall cinemaHall) {
         return cinemaHallDao.add(cinemaHall);
     }
 
+    @Override
+    public CinemaHall get(Long id) {
+        return cinemaHallDao.get(id).orElseThrow(()
+                -> new RuntimeException("There is no such cinema hall"));
```
Comment (Vedroid) [inline src/main/java/cinema/shop/service/impl/CinemaHallServiceImpl.java]: RuntimeException - Less informative than NoSuchElementException

**Review by Vedroid — APPROVED** [review]

File: src/main/java/cinema/shop/controller/MovieSessionController.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    @PutMapping
+    public void update(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession = requestMapper.fromDto(movieSessionRequestDto);
+        movieSessionService.update(movieSession);
+    }
+
+    @DeleteMapping
+    public void delete(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
```
Comment (boroda4436) [inline src/main/java/cinema/shop/controller/MovieSessionController.java]: we should receive an Id of a resource we want to delete
See checklist's link. Try to find an example of `HTTP DELETE http://api.example.com/device-management/managed-devices`

File: src/main/java/cinema/shop/model/dto/request/MovieSessionRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package cinema.shop.model.dto.request;
+
+public class MovieSessionRequestDto {
+    private Long id;
```
Comment (boroda4436) [inline src/main/java/cinema/shop/model/dto/request/MovieSessionRequestDto.java]: How we can use it?

File: src/main/java/cinema/shop/service/impl/CinemaHallServiceImpl.java (hunk 1 above)
Comment (boroda4436) [reply src/main/java/cinema/shop/service/impl/CinemaHallServiceImpl.java]: agree

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/cinema/shop/controller/MovieSessionController.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    public void update(@PathVariable Long id,
+                       @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession = requestMapper.fromDto(movieSessionRequestDto);
+        movieSession.setId(id);
+        movieSessionService.update(movieSession);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id,
+                       @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
```
Comment (boroda4436) [inline src/main/java/cinema/shop/controller/MovieSessionController.java]: Do you still need it?

**Review by boroda4436 — CHANGES_REQUESTED** [review]

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**